db/db_impl/db_impl_write.cc

```diff
@@ -1959,7 +1975,70 @@ void DBImpl::NotifyOnMemTableSealed(ColumnFamilyData* /*cfd*/,
 // REQUIRES: this thread is currently at the front of the 2nd writer queue if
 // two_write_queues_ is true (This is to simplify the reasoning.)
 Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context,
-                              std::string replication_sequence) {
+                              const MemTableSwitchRecord& mem_switch_record) {
+  mutex_.AssertHeld();
+  MemTable* new_mem = nullptr;
+
+  // Recoverable state is persisted in WAL. After memtable switch, WAL might
+  // be deleted, so we write the state to memtable to be persisted as well.
+  Status s = WriteRecoverableState();
+  if (!s.ok()) {
+    return s;
+  }
```

Comment:
Do we care about this, since we don't have a WAL?

Reply:
you are right, we don't need to care about this. This is only updated when we write to WAL